packages/avocode-email-tagsinput/test/e2e/test-utils.js

```diff
@@ -0,0 +1,77 @@
+import { expect } from 'chai'
+
+const tagsInputFieldSelector = '.tagsinput'
+const tagSubmitKeys = [ 'Comma', 'Enter', 'Space' ]
+const URL = isDevelopment
+  ? 'http://localhost:8080/#/avocode-email-tagsinput'
+  : 'https://avocode-email-tagsinput.surge.sh/#/avocode-email-tagsinput'
+
+const getInputText = async (page, inputFieldSelector = tagsInputFieldSelector) => {
+  const text = await page.evaluate((selector) => {
+    const editor = document.querySelector(selector)
+
+    return editor.textContent
+  }, inputFieldSelector)
+
+  return text.trim()
+}
+
+const getTagNodes = async (page, inputFieldSelector = tagsInputFieldSelector) => {
+  return page.evaluate((selector) => {
+    const editor = document.querySelector(selector)
+    const tagNodes = editor.querySelectorAll('.tag')
+
+    return Array.from(tagNodes)
+      .map((node) => {
+        return {
+          text: node.textContent.trim(),
+          className: node.className,
+          html: node.innerHTML,
+        }
+      })
+  }, inputFieldSelector)
+}
+
+
+const getAnchorAndFocus = async (page) => {
+  return page.evaluate(() => {
+    const selection = window.getSelection()
+
+    return [ selection.anchorOffset, selection.focusOffset ]
+  })
+}
+
+const addEmailTag = async (page, emailTag, submitKey = 'Comma') => {
+  await page.click(tagsInputFieldSelector)
+  await page.type(tagsInputFieldSelector, emailTag)
+  if (tagSubmitKeys.includes(submitKey)) {
+    await page.keyboard.press(submitKey)
+  } else {
+    throw new Error(`invalid submit key provided. Expected ${
+      tagSubmitKeys.toString()}, but found: ${submitKey}`
+    )
+  }
+}
+
+const assertEmailTagCount = async function (page, expectedCount) {
```

Comment:
I'm sorry, I know you're trying to save yourself typing the same code over but I really dislike these "assert*" helper methods. I think in the long run they will be more trouble than help.

I think they are hard to read inside the tests. Because when specific test fails, I'm most likely run into the helper method and then I have to go looking into the helper method finding... two lines of code.

Also the `assert...` is weird, I mean the naming is good it's just I'm using `chai` and `jest` here and the naming is to have `expect` calls in the specs. That way I can easily see where the condition is being tested.

So please remove these and just add the code inline. The other helper methods are actually fine and I'm glad you re-used the pattern of testing utilities from `better-react-tagsinput` package 👍